src/utils/logging.js

```diff
@@ -52,61 +45,118 @@ const logToSentry = (event: string | Error, level: SeverityType): string => {
     }
   }
 
-  // The static API's `captureException` doesn't allow passing strings, and its
-  // counterpart `captureMessage` doesn't allow passing stacktraces.
-  // Fortunately, the quasi-internal "Hub" API exists, and is reasonably
-  // well-documented:
-  //
-  // https://docs.sentry.io/development/sdk-dev/unified-api/#hub
-  //
-  // (There is a `captureEvent` method that allows both explicitly; but it also
-  // expects a great deal of other information which we would have to
-  // synthesize, and which has no user-facing documentation.)
-  return getCurrentHub().captureMessage(message, level, hint);
+  return withScope(scope => {
+    scope.setExtras(extras);
+
+    // The static API's `captureException` doesn't allow passing strings, and its
+    // counterpart `captureMessage` doesn't allow passing stacktraces.
+    // Fortunately, the quasi-internal "Hub" API exists, and is reasonably
+    // well-documented:
+    //
+    // https://docs.sentry.io/development/sdk-dev/unified-api/#hub
+    //
+    // (There is a `captureEvent` method that allows both explicitly; but it also
+    // expects a great deal of other information which we would have to
+    // synthesize, and which has no user-facing documentation.)
+    return getCurrentHub().captureMessage(message, level, hint);
+  });
+};
+
+type LogParams = {|
+  consoleMethod: mixed => void,
+  severity: SeverityType,
+|};
+type LogFunction = (event: string | Error, extras?: Extras) => void;
+
+const makeLogFunction = ({ consoleMethod, severity }: LogParams): LogFunction => {
+  const toConsole = consoleMethod.bind(console);
+
+  return (event: string | Error, extras: Extras = {}) => {
+    logToSentry(event, Severity.Error, extras);
+
+    if (config.enableErrorConsoleLogging) {
+      toConsole(event);
+
+      // avoid sending auxiliary data to Sentry twice
+      if (!isSentryActive()) {
```

Comment:
Hmm! What's the story here?

It looks like this means that when Sentry is active, this information will *only* be available in Sentry -- not in `adb logcat` etc. If this information is useful when debugging at the console, then that could be quite annoying when e.g. trying to debug a published release version on someone's device.

Reply:
Having done some digging, I think this deserves some additional testing and discussion which is mostly outside the scope of this PR.

Removed for other, more disappointing reasons.